### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by artursudnik**

### CODE REVIEW

File: apps/zero-api/src/users/users.controller.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @Post()
+  @ApiCreatedResponse({ type: UserEntity })
+  async create(@Body() createUserDto: CreateUserDto) {
+    return await this.usersService.create(createUserDto);
+  }
+
+  // @Get()
```
Comment (arkadiusz-s) [inline apps/zero-api/src/users/users.controller.ts:33]: Please uncomment all this code

**Review by arkadiusz-s — CHANGES_REQUESTED** [review]

File: apps/zero-api/src/interceptors/NoDataInterceptor.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class NoDataInterceptor implements NestInterceptor {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    return next.handle().pipe(
+      map((data) => {
+        // eslint-disable-next-line lodash/prefer-is-nil
+        if (data === null || data === undefined) {
```
[the remark this answers is not in the document]
Comment (artursudnik) [reply apps/zero-api/src/interceptors/NoDataInterceptor.ts]: sure, no problem at all

File: apps/zero-api/src/users/entities/user.entity.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  @ApiProperty({ example: 'John' })
+  firstName: string;
+
+  @ApiProperty({ example: 'Smith' })
+  lastName: string;
+
+  @ApiProperty({ enum: Role, enumName: 'Role'})
+  @IsEnum(Role)
+  role: Role
```
Comment (arkadiusz-s) [inline apps/zero-api/src/users/entities/user.entity.ts:22]: Could You please rename field to "userRole"?

**Review by arkadiusz-s — CHANGES_REQUESTED** [review]

File: apps/zero-api/src/users/entities/user.entity.ts (hunk 3 above)
Comment (artursudnik) [reply apps/zero-api/src/users/entities/user.entity.ts:22]: sure

File: apps/zero-api/src/users/users.controller.ts (hunk 1 above)
Comment (artursudnik) [reply apps/zero-api/src/users/users.controller.ts]: implemented updating user entity, removed all commented out code that is not required now

File: apps/zero-api/src/interceptors/NoDataInterceptor.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor, NotFoundException } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import * as _ from "lodash";
```
Comment (arkadiusz-s) [inline apps/zero-api/src/interceptors/NoDataInterceptor.ts:4]: Please do not use global imports

**Review by arkadiusz-s — APPROVED** [review]

File: apps/zero-api/src/interceptors/NoDataInterceptor.ts (hunk 4 above)
Comment (artursudnik) [reply apps/zero-api/src/interceptors/NoDataInterceptor.ts:4]: importing it as `import isNil from ‘lodash/fp/isNil’` like you suggested causes `isNil_1.default is not a function` errors

**Pull request merged**